In [ ]:
# -*- coding: utf-8 -*-
# 코드 내부에 한글을 사용가능 하게 해주는 부분입니다.

# 딥러닝을 구동하는 데 필요한 케라스 함수를 불러옵니다.
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense

# 필요한 라이브러리를 불러옵니다.
import numpy as np
import tensorflow as tf

# 실행할 때마다 같은 결과를 출력하기 위해 설정하는 부분입니다.
np.random.seed(3)
tf.random.set_seed(3)

# 준비된 수술 환자 데이터를 불러들입니다.
Data_set = np.loadtxt("/content/drive/MyDrive/colab/deeplearning/dataset/ThoraricSurgery.csv", delimiter=",")

# 환자의 기록과 수술 결과를 X와 Y로 구분하여 저장합니다.
X = Data_set[:,0:17]
Y = Data_set[:,17]

# 딥러닝 구조를 결정합니다(모델을 설정하고 실행하는 부분입니다).
model = Sequential()
model.add(Dense(30, input_dim=17, activation='softplus'))
model.add(Dense(1, activation='sigmoid'))

# 딥러닝을 실행합니다.
model.compile(loss='mean_squared_error', optimizer='adam', metrics=['accuracy'])
model.fit(X, Y, epochs=1000, batch_size=10)

Epoch 1/1000
47/47 [==============================] - 1s 2ms/step - loss: 0.2325 - accuracy: 0.7638
Epoch 2/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.1538 - accuracy: 0.8447
Epoch 3/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.1489 - accuracy: 0.8511
Epoch 4/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.1489 - accuracy: 0.8511
Epoch 5/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.1489 - accuracy: 0.8511
Epoch 6/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.1489 - accuracy: 0.8511
Epoch 7/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.1489 - accuracy: 0.8511
Epoch 8/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.1487 - accuracy: 0.8511
Epoch 9/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.1479 - accuracy: 0.8511
Epoch 10/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.1488 - accuracy: 0.8489

In [ ]:
# -*- coding: utf-8 -*-
# 코드 내부에 한글을 사용가능 하게 해주는 부분입니다.

# 딥러닝을 구동하는 데 필요한 케라스 함수를 불러옵니다.
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense

# 필요한 라이브러리를 불러옵니다.
import numpy as np
import tensorflow as tf

# 실행할 때마다 같은 결과를 출력하기 위해 설정하는 부분입니다.
np.random.seed(3)
tf.random.set_seed(3)

# 준비된 수술 환자 데이터를 불러들입니다.
Data_set = np.loadtxt("/content/drive/MyDrive/colab/deeplearning/dataset/ThoraricSurgery.csv", delimiter=",")

# 환자의 기록과 수술 결과를 X와 Y로 구분하여 저장합니다.
X = Data_set[:,0:17]
Y = Data_set[:,17]

# 딥러닝 구조를 결정합니다(모델을 설정하고 실행하는 부분입니다).
model = Sequential()
model.add(Dense(30, input_dim=17, activation='softplus'))
model.add(Dense(1, activation='sigmoid'))

# 딥러닝을 실행합니다.
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])
model.fit(X, Y, epochs=1000, batch_size=10)

Epoch 1/1000
47/47 [==============================] - 1s 2ms/step - loss: 0.6876 - accuracy: 0.8213
Epoch 2/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.5036 - accuracy: 0.8426
Epoch 3/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.4631 - accuracy: 0.8362
Epoch 4/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.5283 - accuracy: 0.8277
Epoch 5/1000
47/47 [==============================] - 0s 3ms/step - loss: 0.4666 - accuracy: 0.8362
Epoch 6/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.4472 - accuracy: 0.8468
Epoch 7/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.4681 - accuracy: 0.8383
Epoch 8/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.4673 - accuracy: 0.8511
Epoch 9/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.4302 - accuracy: 0.8489
Epoch 10/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.4644 - accuracy: 0.8511

In [ ]:
# -*- coding: utf-8 -*-
# 코드 내부에 한글을 사용가능 하게 해주는 부분입니다.

# 딥러닝을 구동하는 데 필요한 케라스 함수를 불러옵니다.
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense

# 필요한 라이브러리를 불러옵니다.
import numpy as np
import tensorflow as tf

# 실행할 때마다 같은 결과를 출력하기 위해 설정하는 부분입니다.
np.random.seed(3)
tf.random.set_seed(3)

# 준비된 수술 환자 데이터를 불러들입니다.
Data_set = np.loadtxt("/content/drive/MyDrive/colab/deeplearning/dataset/ThoraricSurgery.csv", delimiter=",")

# 환자의 기록과 수술 결과를 X와 Y로 구분하여 저장합니다.
X = Data_set[:,0:17]
Y = Data_set[:,17]

# 딥러닝 구조를 결정합니다(모델을 설정하고 실행하는 부분입니다).
model = Sequential()
model.add(Dense(50, input_dim=17, activation='softplus'))
model.add(Dense(1, activation='sigmoid'))

# 딥러닝을 실행합니다.
model.compile(loss='mean_squared_error', optimizer='adam', metrics=['accuracy'])
model.fit(X, Y, epochs=1000, batch_size=10)

Epoch 1/1000
47/47 [==============================] - 1s 2ms/step - loss: 0.7312 - accuracy: 0.2553
Epoch 2/1000
47/47 [==============================] - 0s 3ms/step - loss: 0.1797 - accuracy: 0.8043
Epoch 3/1000
47/47 [==============================] - 0s 4ms/step - loss: 0.1471 - accuracy: 0.8489
Epoch 4/1000
47/47 [==============================] - 0s 4ms/step - loss: 0.1469 - accuracy: 0.8489
Epoch 5/1000
47/47 [==============================] - 0s 3ms/step - loss: 0.1466 - accuracy: 0.8511
Epoch 6/1000
47/47 [==============================] - 0s 3ms/step - loss: 0.1474 - accuracy: 0.8447
Epoch 7/1000
47/47 [==============================] - 0s 3ms/step - loss: 0.1477 - accuracy: 0.8511
Epoch 8/1000
47/47 [==============================] - 0s 3ms/step - loss: 0.1473 - accuracy: 0.8511
Epoch 9/1000
47/47 [==============================] - 0s 3ms/step - loss: 0.1456 - accuracy: 0.8532
Epoch 10/1000
47/47 [==============================] - 0s 3ms/step - loss: 0.1484 - accuracy: 0.8468

In [ ]:
# -*- coding: utf-8 -*-
# 코드 내부에 한글을 사용가능 하게 해주는 부분입니다.

# 딥러닝을 구동하는 데 필요한 케라스 함수를 불러옵니다.
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense

# 필요한 라이브러리를 불러옵니다.
import numpy as np
import tensorflow as tf

# 실행할 때마다 같은 결과를 출력하기 위해 설정하는 부분입니다.
np.random.seed(3)
tf.random.set_seed(3)

# 준비된 수술 환자 데이터를 불러들입니다.
Data_set = np.loadtxt("/content/drive/MyDrive/colab/deeplearning/dataset/ThoraricSurgery.csv", delimiter=",")

# 환자의 기록과 수술 결과를 X와 Y로 구분하여 저장합니다.
X = Data_set[:,0:17]
Y = Data_set[:,17]

# 딥러닝 구조를 결정합니다(모델을 설정하고 실행하는 부분입니다).
model = Sequential()
model.add(Dense(50, input_dim=17, activation='softplus'))
model.add(Dense(30, input_dim=17, activation='relu'))
model.add(Dense(1, activation='sigmoid'))

# 딥러닝을 실행합니다.
model.compile(loss='mean_squared_error', optimizer='adam', metrics=['accuracy'])
model.fit(X, Y, epochs=1000, batch_size=10)

Epoch 1/1000
47/47 [==============================] - 1s 2ms/step - loss: 0.1517 - accuracy: 0.8468
Epoch 2/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.1482 - accuracy: 0.8511
Epoch 3/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.1485 - accuracy: 0.8511
Epoch 4/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.1483 - accuracy: 0.8489
Epoch 5/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.1489 - accuracy: 0.8511
Epoch 6/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.1489 - accuracy: 0.8511
Epoch 7/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.1489 - accuracy: 0.8511
Epoch 8/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.1489 - accuracy: 0.8511
Epoch 9/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.1489 - accuracy: 0.8511
Epoch 10/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.1489 - accuracy: 0.8511

In [1]:
# -*- coding: utf-8 -*-
# 코드 내부에 한글을 사용가능 하게 해주는 부분입니다.

# 딥러닝을 구동하는 데 필요한 케라스 함수를 불러옵니다.
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense

# 필요한 라이브러리를 불러옵니다.
import numpy as np
import tensorflow as tf

# 실행할 때마다 같은 결과를 출력하기 위해 설정하는 부분입니다.
np.random.seed(3)
tf.random.set_seed(3)

# 준비된 수술 환자 데이터를 불러들입니다.
Data_set = np.loadtxt("/content/drive/MyDrive/colab/deeplearning/dataset/ThoraricSurgery.csv", delimiter=",")

# 환자의 기록과 수술 결과를 X와 Y로 구분하여 저장합니다.
X = Data_set[:,0:17]
Y = Data_set[:,17]

# 딥러닝 구조를 결정합니다(모델을 설정하고 실행하는 부분입니다).
model = Sequential()
model.add(Dense(50, input_dim=17, activation='relu'))
model.add(Dense(50, input_dim=17, activation='softplus'))
model.add(Dense(1, activation='sigmoid'))

# 딥러닝을 실행합니다.
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])
model.fit(X, Y, epochs=1000, batch_size=10)

Epoch 1/1000
47/47 [==============================] - 1s 2ms/step - loss: 3.6732 - accuracy: 0.6851
Epoch 2/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.6501 - accuracy: 0.8170
Epoch 3/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.5037 - accuracy: 0.8234
Epoch 4/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.6245 - accuracy: 0.7872
Epoch 5/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.5207 - accuracy: 0.8383
Epoch 6/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.4764 - accuracy: 0.8489
Epoch 7/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.4547 - accuracy: 0.8511
Epoch 8/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.4617 - accuracy: 0.8511
Epoch 9/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.4586 - accuracy: 0.8511
Epoch 10/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.4857 - accuracy: 0.8511

실습1에서 돌린 원본 코드: loss: 0.4018 - accuracy: 0.8468
1. softplus -loss: 0.2878 - accuracy: 0.8915
-> 오차가 줄고 정확도가 늘었다.
2. mean_squared_eroor : loss: 0.1489 - accuracy: 0.8511
-> 오차는 줄었지만 정확도가 떨어졌다.
3. dense 50 : loss: 0.0960 - accuracy: 0.9021
-> 오차가 줄고 정확도가 높아졌다.
4. add() : loss: 0.1489 - accuracy: 0.8511
-> 오차가 늘고 정확도도 떨어졌다.
